subsys/sm_ipt/sm_ipt_os.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "sm_ipt_errno.h"
+#include "sm_ipt_os.h"
+
+#define SM_IPT_LOG_MODULE SM_IPT_OS
+#include "sm_ipt_log.h"
+
+#include <drivers/ipm.h>
+
+#define SHM_NODE            DT_CHOSEN(zephyr_ipc_shm)
+#define SHM_START_ADDR      DT_REG_ADDR(SHM_NODE)
+#define SHM_SIZE            DT_REG_SIZE(SHM_NODE)
+
+void sm_ipt_os_signal(struct sm_ipt_os_ctx *os_ctx)
+{
+	int err = ipm_send(os_ctx->ipm_tx_handle, 0, 0, NULL, 0);
+	if (err != 0) {
+		LOG_ERR("Failed to notify: %d", err);
+	}
+}
+
+void sm_ipt_os_signal_handler(struct sm_ipt_os_ctx *os_ctx, void (*handler)(struct sm_ipt_os_ctx *))
+{
+	os_ctx->signal_handler = handler;
+}
+
+static void ipm_callback(const struct device *ipmdev, void *user_data, uint32_t id,
+			 volatile void *data)
+{
+	struct sm_ipt_os_ctx *os_ctx;
```

Comment:
```suggestion
	struct sm_ipt_os_ctx *os_ctx = (struct sm_ipt_os_ctx *) user_data;
```
Add you can remove next line